### ISSUE TITLE
Error: "Operation returned an invalid status code 'BadRequest' " while using hero-card to show buttons on telegram.

### ISSUE THREAD

**Issue by dhruv2103** [microsoft/BotFramework-Composer#8669]
I am using a Hero Card to display options for multichoice questions. It works fine on webchat(on Azure Portal) but When I deploy on other channels(telegram), it shows me the error.
![image](https://user-images.githubusercontent.com/36305262/133555366-439b56ba-bedc-4ee5-a61d-3de821a45376.png)

I did use multichoice for the first question as shown in the screenshot. I am doing the same thing for one other question, but it is giving me error that "Operation returned an invalid status code 'BadRequest' ".

can anyone help me solve this?

**Comment by dmvtech** [comment]
@dhruv2103 I will take a look. Can you share your HeroCard + prompt? (feel free to sanitize of sensitive info)

**Comment by dmvtech** [comment]
> Can you share your HeroCard + prompt? (feel free to sanitize of sensitive info)

For example, should be something like this:
```json
        {
          "$kind": "Microsoft.ChoiceInput",
          "$designer": {
            "id": "WD52ci"
          },
          "defaultLocale": "en-us",
          "disabled": false,
          "maxTurnCount": 3,
          "alwaysPrompt": false,
          "allowInterruptions": false,
          "unrecognizedPrompt": "",
          "invalidPrompt": "",
          "defaultValueResponse": "",
          "prompt": "${ChoiceInput_Prompt_WD52ci()}",
          "choiceOptions": {
            "includeNumbers": true
          },
          "style": "heroCard",
          "choices": [
            "choice 1",
            "choice 2",
            "choice 3",
            "choice 4"
          ]
        }
```

**Comment by dmvtech** [comment]
I am not able to reproduce the issue, so knowing exactly how your prompt is set up will be essential.

What version of Composer and runtime are you using? C# or Node?

**Comment by dhruv2103** [comment]
I am using a bot framework composer to create a bot in Azure Function (C#). I am following these steps:

Step 1:

- I am creating multi-choice question using ("+" ---> ask a question ---> multi-choice question):
- then I am providing the required information. like,  questions to be asked, options.
- I  chose "heroCard" in list style.


Input:
![1](https://user-images.githubusercontent.com/36305262/133728442-123d463c-8e0b-4ddf-a6a2-297184ffca8d.png)
![2](https://user-images.githubusercontent.com/36305262/133728460-b3e9b676-afe8-463e-8df0-363819f7f6f8.png)

expected output:
![3](https://user-images.githubusercontent.com/36305262/133728644-4a3ed117-5647-4bcf-9ecc-833b628e2727.png)

What I am getting:
![image](https://user-images.githubusercontent.com/36305262/133728705-15626bee-f730-452a-8a6e-6f1d78552a2b.png)


generally, it should work. in fact, it worked for me one choice question. but when I tried to create one more multi-choice question, it did not.

What do you suggest for this?

**Comment by dmvtech** [comment]
Hi @dhruv2103 

A few questions:

- The error just occurs as soon as the choices are rendered, not **_after_** selecting a choice. Correct?
- Is there anything that is happening after the prompt? There's no Send response or anything that could be the cause of the error?
- I see that in the screenshot the property you are using is `user.topic_no_choice` but in the code/json it is showing as  `user.xen_gel_topic`. Is there a discrepancy there? Or is that just two different tests you were doing?

I tried to reproduce by doing as close to what you are, but I am just not hitting the issue. Do you see any errors in your Bot Channels blade in the Azure portal? For example; like I do see for my WebChat:
![image](https://user-images.githubusercontent.com/43397658/133832223-e907b8b5-6b94-423e-82bf-8af896eb98b9.png)

**Comment by dhruv2103** [comment]
Answers to your queries:

- The error just occurs as soon as the choices are rendered, not after selecting a choice. Correct?
     - error occurs as soon as the bot reaches the point where a user should be asked with his/her choice. so, the use-case is something like this. there are main categories and then subcategories. Firstly, the bot asks a user to choose category and then about the subcategory.

- Is there anything that is happening after the prompt? There's no Send response or anything that could be the cause of the error?
     - when I run the flow. The bot is asking users to choose a category using "heroCard" only.
![image](https://user-images.githubusercontent.com/36305262/133957255-0a0dd58b-99be-4e07-9d7d-e49173beb50b.png)
     - as soon as the user selects an option. the bot should be asking subcategories based on users choice for the category. but instead, it is giving an error.
![image](https://user-images.githubusercontent.com/36305262/133957418-a08aa876-9404-4a59-9ac2-c43326c0f71c.png)

- I see that in the screenshot the property you are using is `user.topic_no_choice` but in the code/JSON it is showing as `user.xen_gel_topic`. Is there a discrepancy there? Or is that just two different tests you were doing?
     - it was my mistake. but `user.topic_no_choice` is used to store the category that the user has selected and `user.xen_gel_topic` is for subcategory.
     - but the config is the same for both multi-choice questions.
![image](https://user-images.githubusercontent.com/36305262/133957917-056e202d-029c-4e11-93d2-9ef35a8eb787.png)
     - other cofig is the same as it is in the image that I shared earlier. 

In short, I have 2 multi-choice questions in my bot. I am using heroCard for both of them. for the first choice question, it is working perfectly fine. but for another, it is giving me an error. (`user.xen_gel_topic` is the one in which I am getting an error.)

and lastly, no, I am not getting an error on the portal.
![image](https://user-images.githubusercontent.com/36305262/133958738-fec6595d-58b5-4c93-b340-2e54803d7bda.png)

additionally, It is working when I try my bot in a web chat on the azure portal but is not working in telegram. In telegram, instead of asking a question and giving options, it is giving me an error I mentioned earlier.
![image](https://user-images.githubusercontent.com/36305262/133958995-776481da-a3d9-4dca-be4a-8fa11544844c.png)

**Comment by dmvtech** [comment]
hi @dhruv2103 Thank you. This is certainly interesting.

Is this a production bot (in other channels)? Or are you able to https://docs.microsoft.com/en-us/azure/bot-service/bot-service-debug-channel-ngrok?view=azure-bot-service-4.0 the bot running locally?

Also, these two statements seem a bit contradictory:
> error occurs as soon as the bot reaches the point where a user should be asked with his/her choice

> as soon as the user selects an option. the bot should be asking subcategories based on users choice for the category. but instead, it is giving an error.

Please clarify; the error happens **_after_** choices are presented to user but _**before**_ they actually click on them, correct?

**Comment by dhruv2103** [comment]
Hi,

It is a production bot.

Explanation of two statements that you found contradictory:

as soon as the user selects an option(category). the bot should be asking subcategories based on users choice for the category. but instead, it is giving an error. (Here at this point, a user should be asked with his/her choice  for subcategory but it does not ask for choice rather it is giving me that error.)

I hope now it is clear. if not, write me back.

**Comment by dmvtech** [comment]
Hi @dhruv2103 Any updates? Can you help me understand what you have configured after the prompt, so that I can try and reproduce?

If you can create a dummy bot that can reproduce the issue, then share that with me, that would be the best.
